OpenSim/Actuators/VandenBogert2011Muscle.h

```diff
@@ -0,0 +1,568 @@
+                @returns TBD for now see ImplicitResidual structure*/
+        ImplicitResidual calcImplicitResidual(double muslceLength,
+                                              double projFiberLength,
+                                              double activ,
+                                              double projFiberVelocity,
+                                              double activ_dot,
+                                              double u,
+                                              int returnJacobians) const;
+        /** Calculates the muscle residuals. (State equation form for vector
+        manipulation)
+                @param y vector of:
+                        \li 0: projected fiber length
+                        \li 1: muscle activation
+                @param ydot_guess Vector of the guess for the derivative of y
+                @param muscleLength length of muscle (N)
+                @param u muscle control (excitation)
+                @param returnJacobians if=1, calculate and return the
+                        Jacobains. default is =0, no Jacobians not calculated
+                @returns TBD for now see ImplicitResidual structure*/
+        ImplicitResidual calcImplicitResidual(SimTK::Vec2 y,
+                                              SimTK::Vec2 ydot_guess,
+                                              double muscleLength,
+                                              double u,
+                                              int returnJacobians) const;
+
+        /** Calculates the muscle residuals. (method for getting from
+        state)
+                @param s The state of the system. (provides projected fiber
+                        length and muscle activation)
+                @param projFibVel_guess Guess of the fiber velocity
+                        (projected in-line with the tendon)
+                @param activdot_guess Guess of the time derivative of activation
+                @param u muscle control (excitation)
+                @param returnJacobians if=1, calculate and return the
+                        Jacobains. default is =0, no Jacobians not calculated
+                @returns TBD for now see ImplicitResidual structure*/
+        ImplicitResidual calcImplicitResidual(const SimTK::State& s,
+                                              double projFibVel_guess,
+                                              double activdot_guess,
+                                              double u,
+                                              int returnJacobians);
+
+        //Convience methods for converting between muscle length, projected
+        // muscle length, and normalized values.  Need to do this because I am
+        // sticking with the convention that external to the muscle,
+        // muscle length is the state variable.  Inside this muscle's code it
+        // is using the projected muscle length.
```

Comment:
Incorporated(ing).  My reason for making it externally being actual length was so that it matched other muscle and  in case  that was assumed by other methods.  But I agree that we should just use it in the state also and just make sure it is know that it is not muscle length.  Thx.